dynamic_rest/client.py

```diff
@@ -0,0 +1,196 @@
+        self.password = password
+        self.token = token
+        self.scheme = scheme
+        self.authorization_type = authorization_type
+
+    @property
+    def token(self):
+        return self._token
+
+    @token.setter
+    def token(self, value):
+        self._token = value
+        self.authenticated = bool(value)
+        if value:
+            self.session.headers.update({
+                'Authorization': '%s %s' % (
+                    self.AUTHORIZATION_TYPE,
+                    self._token
+                )
+            })
+
+    @property
+    def sessionid(self):
+        return self._sessionid
+
+    @sessionid.setter
+    def sessionid(self, value):
+        self._sessionid = value
+        self.authenticated = bool(value)
+        if value:
+            self.session.headers.update({
+                'Cookie': 'sessionid=%s' % value
+            })
+
+    def authenticate(self, raise_exception=True):
+        response = None
+        if not self.authenticated:
+            username = self.username
+            password = self.password
+            response = requests.post(
+                self.build_url(API_AUTH_ENDPOINT),
+                data={
+                    'login': username,
+                    'password': password
+                },
+                allow_redirects=False
+            )
+            if raise_exception:
+                response.raise_for_status()
+
+            self.sessionid = response.cookies.get('sessionid')
+
+        if raise_exception and not self.authenticated:
+            raise AuthenticationFailed(
+                response.text if response else 'Unknown error'
+            )
+        return self.authenticated
+
+    def build_url(self, url, prefix=None):
+        if not url.startswith('/'):
+            url = '/%s' % url
+
+        if prefix:
+            if not prefix.startswith('/'):
+                prefix = '/%s' % prefix
+
+            url = '%s%s' % (prefix, url)
+        return '%s://%s%s' % (self.scheme, self.host, url)
+
+    def save(
+        self,
+        resource,
+        data,
+        **kwargs
+    ):
+        if 'id' in data:
+            update = True
+            pk = data['id']
+        else:
+            update = False
+            pk = None
+
+        data = json.dumps(data)
+        url = self.get_resource_url(resource, pk)
+        try:
+            verb = 'put' if update else 'post'
+            response = getattr(self, verb)(url, data=data)
+            return response
+        except requests.exceptions.HTTPError as e:
+            return e.response
+
+    def get_resource_body(self, response, resource, many=False):
+        body = json.loads(response.text)
+        key = resource if many else inflection.singularize(resource)
+        return body[key]
+
+    def get_resource_url(self, resource, pk=None):
+        return '%s/%s' % (
+            resource,
+            '%s/' % pk if pk else ''
+        )
+
+    def find(
```

Comment:
Hm... This might be more work, but how do you feel about a `QuerySet` like interface? Something that would look like:

```python
Resource.filter(foo=bar).include('field1', 'field2')
```